### PR TITLE
Improvements to One-Shot user-guide

### PR DESCRIPTION
**Pull request by fareeha**

### CODE REVIEW

File: userguide/one_shot_object_detection/README.md [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+One-Shot object detection is the task of detecting an object from as few as one example per category. Unlike the Object Detector which requires many varied examples of objects in the real world, the One-Shot detector only requires a single canonical example of an object to train the classifier on and performs predictions for that category in the wild. It is best suited for 2D objects or when there isn't much expected variance in the examples, such as playing cards, logos, road signs, and clapperboards. One-Shot detector is not suitable for 3D objects such as faces, animals, and cars. 
 
-Given an image and a one-shot detector trained on category **stop sign**, the output prediction will look like:
+Given a starter image (*left*), a one-shot detector trained on this starter image will produce instance predictions on a test image (*right*) which may look like this:
```
Comment (znation) [inline userguide/one_shot_object_detection/README.md]: Where are the left/right images? I only see one image linked below.

**Comment by znation** [comment]
Just one comment, otherwise this looks good to me!

File: userguide/one_shot_object_detection/README.md (hunk 1 above)
Comment (fareeha) [reply userguide/one_shot_object_detection/README.md]: I stitched the two separate images (starter image of stop sign and the test image showing the predictions) together because there wasn't a clean way of making sure the images aligned. I see the confusion in that sentence now. Is there a better way for us to point to the two separately within the same image?

Comment (znation) [reply userguide/one_shot_object_detection/README.md]: Oh, I didn't see that. I think stitching them together is fine.

File: userguide/one_shot_object_detection/README.md [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 
 ![Stop Sign vector and prediction images](images/stop_sign_sample.jpg) 
 
 
 #### Introductory Example
 
-In this example, the goal is to **localize** where in the image instances of the category **stop sign** are present. For this task we supply a single image of a stop sign as a starter image:
+In this example, our goal is to **localize** instances of **stop sign** in images. For this application, we supply a single image of a stop sign as a starter image:
```
Comment (srikris) [inline userguide/one_shot_object_detection/README.md]: To train a model for this application

File: userguide/one_shot_object_detection/README.md [hunk 3]
```diff
@@ -27,15 +27,15 @@ test_data = tc.SFrame({'image':[tc.Image('stop_sign_photo1.png'),
...
 test_data['image_with_predictions'] = \
 	tc.object_detector.util.draw_bounding_boxes(test_data['image'], test_data['predictions']) 
 
-# Evaluate the model and save the results in a dictionary
-metrics = model.evaluate(test_data)
+# To visualize the predictions made on the test set
+test_data.explore()
 
 # Save the model for later use in TuriCreate
 model.save('mymodel.model')
```
Comment (srikris) [inline userguide/one_shot_object_detection/README.md]: Use another name other than my model and my detector. Something like 'MyStopSignDetector'

File: userguide/one_shot_object_detection/README.md [hunk 4]
```diff
@@ -44,34 +44,24 @@ model.save('mymodel.model')
 model.export_coreml('MyCustomOneShotDetector.mlmodel')
 ```
 
-Previewing the synthetic images generated for training is useful to gauge the kind of test images the model would be trained to handle. We can preview the augmented images generated and used by the model for training as follows:  
+Before starting the training process, you can preview the synthetic images the toolkit generates by calling the `preview_synthetic_training_data` utility. This helps you understand data the model would be trained on:
```
Comment (srikris) [inline userguide/one_shot_object_detection/README.md]: I would move this part to after how it works. It lacks context otherwise.

File: userguide/one_shot_object_detection/README.md [hunk 5]
```diff
@@ -82,7 +72,9 @@ Examples of test image predictions:
 
 #### How it works
 
-To begin training for one-shot object detector, a set of synthetic training data needs to be generated using the provided starter training image(s). These images are augmented by applying yaw, pitch, and roll rotations. These operations are applied on a set of background images. The background images may be provided by the user. If no background images are provided, a set of default background images are automatically downloaded and used.
+To begin training a one-shot object detector, a set of synthetic training data needs to be generated using the provided starter training image(s). The synthetic images are generated by applying yaw, pitch, and roll rotations on a set of background images. The background images may be provided by the user. If no background images are provided, a set of default background images are automatically downloaded and used.
```
Comment (srikris) [inline userguide/one_shot_object_detection/README.md]: I think you need to simplify this explanation. 

In this section, we will go into detail on what happens under the hood of the create method. An OSOD model is trained in 2 steps. First, we use the starter images to generate synthetic data and annotations, and second, we train an object detector.

Then go into a short paragraph explaining how the synthetic images are generated.

File: userguide/one_shot_object_detection/README.md [hunk 6]
```diff
@@ -96,7 +88,7 @@ To learn more about deploying One-Shot to CoreML details, refer to the [Object D
 
 **Background images**
```
Comment (srikris) [inline userguide/one_shot_object_detection/README.md]: I would add a section called Advanced usage and then show this + some of the other things in the user guide.

File: userguide/one_shot_object_detection/README.md [hunk 7]
```diff
@@ -109,16 +101,14 @@ starter_images = tc.SFrame({'image':[tc.Image('stop_sign.png')],
 my_backgrounds = tc.SArray('my_custom_backgrounds.sarray')
 
 # Create a model using custom background images                                    
-model = tc.one_shot_object_detector.create(starter_images, 'label', backgrounds = my_backgrounds)
+model = tc.one_shot_object_detector.create(starter_images, 'label', backgrounds=my_backgrounds)
 ```
 
-To view the synthetic images generated with your custom background images:
+To preview the synthetic images generated with your custom background images:
```
Comment (srikris) [inline userguide/one_shot_object_detection/README.md]: Repeated synthetic images section. I'd show it only once.

File: userguide/one_shot_object_detection/README.md (hunk 2 above)
Comment (srikris) [inline userguide/one_shot_object_detection/README.md]: Can simplify this sentence to read more tight.

File: userguide/one_shot_object_detection/README.md (hunk 3 above)
Comment (fareeha) [reply userguide/one_shot_object_detection/README.md]: I was following the model naming convention in the other toolkit user-guides. Apart from DC and Text classifier, all toolkits save the model as `mymodel.model` and the coreml model as `MyCustom***.mlmodel`. Should I care about following this convention?

File: userguide/one_shot_object_detection/README.md (hunk 4 above)
Comment (fareeha) [reply userguide/one_shot_object_detection/README.md]: Shall I move it in 'How it works'? Since we talk about generating synthetic images in that.

If we move to the next relevant section after it, that would be 'Advanced Usage - background images' where we talk about using custom background images and displaying our background images might be a bit confusing.

File: userguide/one_shot_object_detection/README.md (hunk 6 above)
Comment (fareeha) [reply userguide/one_shot_object_detection/README.md]: This section 'Background images' is under the section 'Advanced Usage'. Is there anything else you feel we should add to Advanced Usage?

File: userguide/one_shot_object_detection/README.md (hunk 7 above)
Comment (fareeha) [reply userguide/one_shot_object_detection/README.md]: The difference here is the added third argument which is `my_backgrounds` when calling `preview_synthetic_training_data`. I thought it might be helpful to show the change in code needed to preview synthetic images when using custom backgrounds. I was thinking it's better to be thorough over brevity.

File: userguide/one_shot_object_detection/README.md (hunk 3 above)
Comment (srikris) [reply userguide/one_shot_object_detection/README.md]: You can leave it as is.

File: userguide/one_shot_object_detection/README.md (hunk 4 above)
Comment (srikris) [reply userguide/one_shot_object_detection/README.md]: You can move it to how it works. In that case, you don't need an advanced usage section.

File: userguide/one_shot_object_detection/README.md [hunk 8]
```diff
@@ -1,15 +1,15 @@
...
 
-One-Shot object detection is the task of learning to detect from as few as one example per category. Unlike the Object Detector which requires many varied examples of objects in the real world, the One-Shot detector only requires a single canonical example of an object to train the classifier on and performs predictions for that category in the wild. It is best suited for 2D objects or when there isn't much expected variance in the examples, such as playing cards, logos, road signs and clapperboards. One-Shot detector is not suitable for 3D objects such as faces, animals and cars. 
+One-Shot object detection (OSOD) is the task of detecting an object from as few as **one** example per category. Unlike the Object Detector which requires many varied examples of objects in the real world, the One-Shot detector only requires a single canonical example of an object to train the classifier on and perform predictions for that category in the wild. It is best suited for 2D objects or when there isn't much expected variance in the examples, such as playing cards, logos, road signs, and clapperboards. One-Shot detector is not suitable for 3D objects such as faces, animals, and cars.
```
Comment (srikris) [inline userguide/one_shot_object_detection/README.md]: unlike the object detector (add link). Also clarify that for 3D objects, they should use the regular OD.

File: userguide/one_shot_object_detection/README.md [hunk 9]
```diff
@@ -1,15 +1,15 @@
...
+One-Shot object detection (OSOD) is the task of detecting an object from as few as **one** example per category. Unlike the Object Detector which requires many varied examples of objects in the real world, the One-Shot detector only requires a single canonical example of an object to train the classifier on and perform predictions for that category in the wild. It is best suited for 2D objects or when there isn't much expected variance in the examples, such as playing cards, logos, road signs, and clapperboards. One-Shot detector is not suitable for 3D objects such as faces, animals, and cars. 
 
-Given an image and a one-shot detector trained on category **stop sign**, the output prediction will look like:
+Given a starter image (*left*), a one-shot detector trained on this starter image will produce instance predictions on a test image (*right*) which may look like this:
```
Comment (srikris) [inline userguide/one_shot_object_detection/README.md]: I would move this sentence to below the intro example because it's easier to parse once you know you are trying to make a stop sign detector.

File: userguide/one_shot_object_detection/README.md [hunk 10]
```diff
@@ -1,15 +1,15 @@
...
 
 ![Stop Sign vector and prediction images](images/stop_sign_sample.jpg) 
 
 
 #### Introductory Example
 
-In this example, the goal is to **localize** where in the image instances of the category **stop sign** are present. For this task we supply a single image of a stop sign as a starter image:
+In this example, our goal is to **localize** instances of **stop sign** in images. To train a model for this application, we supply a single image of a stop sign as a starter image:
```
Comment (srikris) [inline userguide/one_shot_object_detection/README.md]: stop signs

File: userguide/one_shot_object_detection/README.md [hunk 11]
```diff
@@ -44,81 +44,67 @@ model.save('mymodel.model')
...
-To visualize the predictions made on the test set:
-
-
-```python
-test_data['image_with_predictions'] = \
-	tc.object_detector.util.draw_bounding_boxes(test_data['image'], test_data['predictions'])
-
-test_data.explore()
-```
+Visualizing predictions made on the test set:
```
Comment (srikris) [inline userguide/one_shot_object_detection/README.md]: I don't think this explore on the test set is adding much value. I'd leave remove it.

File: userguide/one_shot_object_detection/README.md [hunk 12]
```diff
@@ -44,81 +44,67 @@ model.save('mymodel.model')
...
 ![Sample prediction image - 1 and 2](images/sample_prediction_images_1_2.jpg)
 
 
 #### How it works
 
-To begin training for one-shot object detector, a set of synthetic training data needs to be generated using the provided starter training image(s). These images are augmented by applying yaw, pitch, and roll rotations. These operations are applied on a set of background images. The background images may be provided by the user. If no background images are provided, a set of default background images are automatically downloaded and used.
+In this section, we will go into detail on what happens under the hood of the `create` method. An OSOD model is trained in two steps: First, we use the starter images to generate synthetic data and annotations, and second, we train an object detector.
 
-To learn more about the training implementation details, refer to the [Object Detector: How it Works](https://apple.github.io/turicreate/docs/userguide/object_detection/how-it-works.html) chapter.
+To begin training, a set of synthetic training data needs to be generated using the provided starter image(s).
```
Comment (srikris) [inline userguide/one_shot_object_detection/README.md]: I would title this as **Stage 1: Automatic synthetic data generation** and then explain this. This paragraph does't make it clear that his is happening under the hood.

File: userguide/one_shot_object_detection/README.md [hunk 13]
```diff
@@ -44,81 +44,67 @@ model.save('mymodel.model')
...
+In this section, we will go into detail on what happens under the hood of the `create` method. An OSOD model is trained in two steps: First, we use the starter images to generate synthetic data and annotations, and second, we train an object detector.
 
-To learn more about the training implementation details, refer to the [Object Detector: How it Works](https://apple.github.io/turicreate/docs/userguide/object_detection/how-it-works.html) chapter.
+To begin training, a set of synthetic training data needs to be generated using the provided starter image(s). 
+The synthetic images are generated by applying yaw, pitch, and roll rotations on a set of background images. The background images may be provided by the user. If no background images are provided, a set of default background images are automatically downloaded and used.
```
Comment (srikris) [inline userguide/one_shot_object_detection/README.md]: You want to invert this statement. A set of default backgrounds are automatically downloaded and they can be set to a user specified backgrounds to better tune the model to the user scenario.

File: userguide/one_shot_object_detection/README.md [hunk 14]
```diff
@@ -44,81 +44,67 @@ model.save('mymodel.model')
...
 
+You can preview the synthetic images the toolkit generates by calling the `preview_synthetic_training_data` utility. This helps you understand data the model would be trained on:  
 
-#### Deployment to CoreML
 
-To learn more about deploying One-Shot to CoreML details, refer to the [Object Detector: Deployment to Core ML](https://apple.github.io/turicreate/docs/userguide/object_detection/export-coreml.html) chapter.
+```python
```
Comment (srikris) [inline userguide/one_shot_object_detection/README.md]: On reading this, I am more convinced that we should have an advanced section that focusses only on the backgrounds and a place to add more things in the future. The first exposure (intro, how it works) should be very simple.  The how-it-works section seems to go back and forth between explaining how it works and explaining how to use more advanced options.

File: userguide/one_shot_object_detection/README.md [hunk 15]
```diff
@@ -1,15 +1,16 @@
...
+One-Shot object detection (OSOD) is the task of detecting an object from as few as **one** example per category. Unlike the [Object Detector](https://apple.github.io/turicreate/docs/userguide/object_detection) which requires many varied examples of objects in the real world, the One-Shot detector only requires a single canonical example of an object to train the classifier on and perform predictions for that category in the wild. It is best suited for 2D objects or when there isn't much expected variance in the examples, such as playing cards, logos, road signs, and clapperboards. One-Shot detector is not suitable for 3D objects such as faces, animals, and cars which is a task better suited for the regular [Object Detector](https://apple.github.io/turicreate/docs/userguide/object_detection). 
 
-Given an image and a one-shot detector trained on category **stop sign**, the output prediction will look like:
 
-![Stop Sign vector and prediction images](images/stop_sign_sample.jpg) 
 
+## Introductory Example
+
+Given a starter image (*left*), a one-shot detector trained on this starter image will produce instance predictions on a test image (*right*) which may look like this:
```
Comment (srikris) [inline userguide/one_shot_object_detection/README.md]: Swap the paragraph here and the one below it?

File: userguide/one_shot_object_detection/README.md [hunk 16]
```diff
@@ -44,81 +45,70 @@ model.save('mymodel.model')
...
 
-```python
-test_data['image_with_predictions'] = \
-	tc.object_detector.util.draw_bounding_boxes(test_data['image'], test_data['predictions'])
 
-test_data.explore()
-```
 
-![Prediction test data](images/test_data_explore.jpg)
+## Deployment to CoreML
```
Comment (srikris) [inline userguide/one_shot_object_detection/README.md]: Why is this in this section. Shouldn't this be in the intro example?

Comment (fareeha) [reply userguide/one_shot_object_detection/README.md]: Oh good point, I placed it under the intro section.
